planner/core/find_best_task.go

```diff
@@ -1124,17 +1128,40 @@ func (ds *DataSource) buildIndexMergeTableScan(prop *property.PhysicalProperty,
 		ts.stats.StatsVersion = statistics.PseudoVersion
 	}
 	if len(tableFilters) > 0 {
+		// Here we want to add a Selection for exprs that cannot be pushed to TiKV, but some of them has already been put
+		// in Selection above DataSource, so we need to filter these exprs.
+		_, filtersInSelection := expression.PushDownExprs(sessVars.StmtCtx, tableFilters, ds.ctx.GetClient(), kv.UnSpecified)
```

Comment:
extract a function to extract the functions which can not be pushed down to tikv but  can be pushed down to other storages.

Reply:
done